### DISCUSSION TITLE
Request for BlockStateTag in kits.

### DISCUSSION THREAD

**Discussion by cataclisto** [EssentialsX/Essentials#4360] [Ideas and Feature Suggestions] [answered]
Right now the current kits do not save blockstates on held items. If I were to spawn in the light blocks 1-15 The blocks would be separate in my inventory, but if a kit would be made with them it would only save the default light block and give you 15 level 15's when a kit is given. 
A few examples of the mentioned ids if you were to spawn them in...
/item light 1 {BlockStateTag: {level:"1"}}
/item light 1 {BlockStateTag: {level:"2"}}

**Comment by triagonal** [comment, marked as answer]
By default, the `/createkit` command is unable to record NBT data on items. You have two options:

1. Add the NBT data manually to items in the kits.yml file in the format `- itemname amount {NBT...}`. Note that if you choose this option, you can't also use essentials meta such as `name:` or `lore:` on the same item.
2. If you're running Paper on 1.15.2+, you can use the new `use-nbt-serialization-in-createkit` which *will* record item NBT data when using `/createkit`, but will store the kit items in a non human-editable format inside the kits.yml file.

**Reply by cataclisto** [reply]
Ahh alright, thank you for mentioning that! There is not much documentation mentioning nbt serialization in paper or essentials (And I didn't see it in default essentials dev config or the paper config, could of missed it, unsure). Using Google to search for that line didn't help much either as only one single result returned and it was a response to a comment similiar to mine.

**Reply by triagonal** [reply]
My bad, the option is in the Essentials config. I thought I had typed that but I must have left it out. It's only present in dev builds and the config doesn't update automatically, so if you don't have it you can just add it yourself. 

https://github.com/EssentialsX/Essentials/blob/1556b911b23524a12d4816313bc98c950149cfa4/Essentials/src/main/resources/config.yml#L337-L344
